docs/performance.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Performance benchmarks for v1.15.0
+permalink: /performance/
+layout: default
+---
+
+# {{ page.title }}
+
+These are the execution times for the current indicators using two years of historical daily stock quotes (502 periods) with default or typical parameters.
+
+``` bash
+BenchmarkDotNet=v0.13.0, OS=Windows 10.0.19043.1165 (21H1/May2021Update)
+Intel Core i7-7700HQ CPU 2.80GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
+.NET SDK=5.0.303
+  [Host]     : .NET 5.0.9 (5.0.921.35908), X64 RyuJIT
+  DefaultJob : .NET 5.0.9 (5.0.921.35908), X64 RyuJIT
+```
```

Comment:
Do you have any specific system requirements for benchmarking?

Reply:
No, but we should be consistent when updating them as they will vary from one processor over another.  I think the more "normal" it is the better as anyone can make them go fast with a supercomputer.

Reply:
Does this look about right to you?  I wonder if we're measuring history loading time too.  Seems a bit high.  We can take a closer look later; after we get an initial package published.

```bash
Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
-------------------------------------------------------------------------------------- benchmark: 18 tests ---------------------------------------------------------------------------------------
Name (time in ms)                     Min                 Max              Mean             StdDev            Median               IQR            Outliers       OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_benchmark_fractal             1.5812 (1.0)        5.1889 (1.84)     2.1103 (1.11)      0.7300 (3.64)     1.7801 (1.0)      0.5202 (3.31)        21;19  473.8666 (0.90)        198           1
test_benchmark_ema                 1.7112 (1.08)       2.8249 (1.0)      1.8976 (1.0)       0.2008 (1.0)      1.8071 (1.02)     0.1572 (1.0)         22;16  526.9838 (1.0)         139           1================
test_benchmark_sma                 1.7176 (1.09)       3.8023 (1.35)     2.0684 (1.09)      0.3796 (1.89)     1.9127 (1.07)     0.4325 (2.75)        62;19  483.4762 (0.92)        410           1
test_benchmark_adl                 1.7653 (1.12)       4.3929 (1.56)     1.9536 (1.03)      0.3159 (1.57)     1.8547 (1.04)     0.1788 (1.14)          6;6  511.8743 (0.97)         87           1
test_benchmark_atr                 1.7823 (1.13)       4.4623 (1.58)     1.9900 (1.05)      0.3766 (1.88)     1.8610 (1.05)     0.2280 (1.45)        27;24  502.5008 (0.95)        386           1
test_benchmark_alma                1.8751 (1.19)     110.4235 (39.09)    2.7768 (1.46)      8.1429 (40.56)    1.9457 (1.09)     0.2471 (1.57)         1;15  360.1318 (0.68)        178           1
test_benchmark_alligator           2.0334 (1.29)       8.3589 (2.96)     2.5431 (1.34)      1.0612 (5.29)     2.1381 (1.20)     0.2430 (1.55)        11;16  393.2282 (0.75)        116           1
test_benchmark_aroon               2.0503 (1.30)       6.2525 (2.21)     2.3467 (1.24)      0.6126 (3.05)     2.1612 (1.21)     0.2499 (1.59)         8;11  426.1259 (0.81)        182           1
test_benchmark_awesome             2.0748 (1.31)       3.4522 (1.22)     2.2294 (1.17)      0.2142 (1.07)     2.1313 (1.20)     0.1800 (1.15)         22;9  448.5511 (0.85)        157           1          [100%]
test_benchmark_rsi                 2.1288 (1.35)       4.1079 (1.45)     2.5431 (1.34)      0.3734 (1.86)     2.4402 (1.37)     0.4506 (2.87)         29;6  393.2260 (0.75)        146           1
test_benchmark_super_trend         2.1786 (1.38)       7.2250 (2.56)     2.8826 (1.52)      0.8535 (4.25)     2.6125 (1.47)     0.7309 (4.65)        18;11  346.9149 (0.66)        146           1
test_benchmark_macd                2.1862 (1.38)       4.0266 (1.43)     2.6765 (1.41)      0.4012 (2.00)     2.5650 (1.44)     0.5403 (3.44)         41;3  373.6235 (0.71)        142           1
test_benchmark_bollinger_bands     2.2619 (1.43)     109.1363 (38.63)    3.3865 (1.78)      8.9990 (44.82)    2.4073 (1.35)     0.3851 (2.45)          1;8  295.2893 (0.56)        141           1
test_benchmark_stoch               2.3917 (1.51)       4.8807 (1.73)     2.8856 (1.52)      0.4650 (2.32)     2.7334 (1.54)     0.5885 (3.74)         51;8  346.5462 (0.66)        253           1
test_benchmark_stoch_rsi           3.0193 (1.91)     115.8500 (41.01)    4.8506 (2.56)     11.0599 (55.09)    3.6106 (2.03)     0.9586 (6.10)          1;1  206.1586 (0.39)        103           1
test_benchmark_adx                 3.0727 (1.94)       5.1545 (1.82)     3.3092 (1.74)      0.2914 (1.45)     3.2023 (1.80)     0.2619 (1.67)          9;4  302.1876 (0.57)         81           1
test_benchmark_beta                4.4462 (2.81)      11.7370 (4.15)     5.0732 (2.67)      1.0437 (5.20)     4.8536 (2.73)     0.4798 (3.05)          4;4  197.1150 (0.37)         90           1
test_benchmark_sma_extended        5.0935 (3.22)      18.2127 (6.45)     7.4280 (3.91)      2.6923 (13.41)    6.3497 (3.57)     2.8261 (17.97)        10;5  134.6261 (0.26)         82           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
=============================================================================================== 18 passed in 9.64s ===============================================================================================
```

Reply:
Do you think Python.NET will support newer .NET editions soon, in v3.0?  Looks like it will.  Newer editions, like .NET Core 3.1, 5.0, and 6.0 are measurably faster than .NET Standard.  No need to wait for it, but I suspect this will probably cut these times in half.

Reply:
> No, but we should be consistent when updating them as they will vary from one processor over another. I think the more "normal" it is the better as anyone can make them go fast with a supercomputer.

Okay, then I'll use my laptop for benchmarking.

> Does this look about right to you? I wonder if we're measuring history loading time too. Seems a bit high. We can take a closer look later; after we get an initial package published.

Yes, you're right. It's too long compared to C# implementation. I don't think we're measuring history loading time , because the data is loaded at the start of the test at once and is re-used for each test.
Sadly, this result maybe make sense, according to this issue:
 - https://github.com/pythonnet/pythonnet/issues/694

> Do you think Python.NET will support newer .NET editions soon, in v3.0? Looks like it will. Newer editions, like .NET Core 3.1, 5.0, and 6.0 are measurably faster than .NET Standard. No need to wait for it, but I suspect this will probably cut these times in half.

I'm not sure when they release pythonnet v3.0 but guessing that will be next year. The milestone for v3.0 is still indicating 68%(30~40% 2months ago)
Also, I think that supporting newer .NET edition won't give us huge improvement, C# implementation(using as DLL) is already fast enough, the problem is coming from converting C# to Python.

We should try more and can manage it as an issue.